packages/picker/src/Picker.ts

```diff
@@ -335,7 +350,7 @@ export class PickerBase extends SizedMixin(Focusable, { noDefaultSize: true }) {
     }
 
     public toggle(target?: boolean): void {
-        if (this.readonly || this.pending) {
+        if (this.readonly || this.pendingStateController.isPending()) {
```

Comment:
Can we not eliminate these `isPending()` function calls and instead manage a simple variable in the `pendingStateController`?

Reply:
The isPending function offers dynamic flexibility in determining the pending state. Directly managing a variable to store this state would reduce this flexibility. Additionally, introducing an extra state variable requires meticulous synchronization with the host’s state changes. This imposes additional and potentially redundant responsibilities on the controller. If the state variable is not accurately updated or if certain edge cases are not handled, it can lead to inconsistencies in the UI. Therefore, maintaining the isPending function ensures a more robust and flexible approach to managing the pending state.

Reply:
a function call is an overhead and it is memory intensive. Why don't you create a property called isPending in the controller and access it here?